hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestOMStorage.java

```diff
@@ -17,76 +17,312 @@
+    assertGetNodeProperties(storage, OM_ID_STR, nodeId);
+  }
+
+  @Test
+  public void testValidateOrPersistOmNodeIdDoesRunWithSameNodeIdAsInFile()
+      throws Exception {
+    String nodeId = "nodeId";
+    OzoneConfiguration conf = configWithOMDBDir();
+    setupAPersistedVersionFileWithNodeId(conf, nodeId);
+
+    OMStorage storage = new OMStorage(conf);
+    assertThat(storage.getState(), is(INITIALIZED));
+    assertThat(storage.getOmNodeId(), is(nodeId));
+    assertGetNodeProperties(storage, OM_ID_STR, nodeId);
+
+    storage.validateOrPersistOmNodeId(nodeId);
+
+    assertThat(storage.getOmNodeId(), is(nodeId));
+    assertGetNodeProperties(storage, OM_ID_STR, nodeId);
+  }
+
+  @Test
+  public void testValidateOrPersistOmNodeIdThrowsWithDifferentNodeIdAsInFile()
+      throws Exception {
+    String nodeId = "nodeId";
+    String newId = "newId";
+    OzoneConfiguration conf = configWithOMDBDir();
+    setupAPersistedVersionFileWithNodeId(conf, nodeId);
+
+    OMStorage storage = new OMStorage(conf);
+    assertThat(storage.getState(), is(INITIALIZED));
+    assertThat(storage.getOmNodeId(), is(nodeId));
+
+    thrown.expect(IOException.class);
+    String expectedMsg =
+        String.format(ERROR_UNEXPECTED_OM_NODE_ID_TEMPLATE, nodeId, newId);
+    thrown.expectMessage(expectedMsg);
+
+    storage.validateOrPersistOmNodeId(newId);
+  }
+
+  private void assertCertOps(OMStorage storage) throws IOException {
+    String certSerialId = "12345";
+    String certSerialId2 = "54321";
+    storage.setOmCertSerialId(certSerialId);
+    assertThat(storage.getOmCertSerialId(), is(certSerialId));
+    assertGetNodeProperties(storage, null, null, certSerialId);
+
+    storage.setOmCertSerialId(certSerialId2);
+    assertThat(storage.getOmCertSerialId(), is(certSerialId2));
+    assertGetNodeProperties(storage, null, null, certSerialId2);
+
+    storage.unsetOmCertSerialId();
+    assertThat(storage.getOmCertSerialId(), is(nullValue()));
+    assertGetNodeProperties(storage, null, null, null);
+  }
+
+  private void assertGetNodeProperties(OMStorage storage, String... values) {
+    Properties p = storage.getNodeProperties();
+    Map<String, String> e = toExpectedPropertyMapping(values);
+
+    if (e.get(OM_ID) != null) {
+      assertThat(p.getProperty(OM_ID), is(e.get(OM_ID)));
+    }
+    if (e.get(OM_NODE_ID) != null) {
+      assertThat(p.get(OM_NODE_ID), is(e.get(OM_NODE_ID)));
+    }
+    if (e.get(OM_CERT_SERIAL_ID) != null) {
+      assertThat(p.get(OM_CERT_SERIAL_ID), is(e.get(OM_CERT_SERIAL_ID)));
+    }
+  }
+
+  private Map<String, String> toExpectedPropertyMapping(String... values) {
+    Map<String, String> ret = new HashMap<>();
+    String[] propNames = new String[]{OM_ID, OM_NODE_ID, OM_CERT_SERIAL_ID};
+    for (int i = 0; i < values.length; i++) {
+      ret.put(propNames[i], values[i]);
+    }
+    return ret;
+  }
+
+  private OMStorage setupAPersistedVersionFile(OzoneConfiguration conf)
```

Comment:
Looks like the return of this method is never used. Should it be removed or is there a use case for the return value that got missed?

Reply:
First it was used, then I realized that an OMStorage object never gets initialized just if it is being created with a config that points to a directory that contains a proper VERSION file, after that the return value was not necessary anymore, but I did not realized this. Thank you for recognizing this, I am changing the return value to void.